core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3501,7 +3501,7 @@ public PhysicalOperation visitExchange(ExchangeNode node, LocalExecutionPlanCont
         private boolean isLocalScaledWriterExchange(PlanNode node)
         {
             Optional<PlanNode> result = searchFrom(node)
-                    .where(planNode -> node instanceof ExchangeNode && ((ExchangeNode) node).getScope() == LOCAL)
+                    .where(planNode -> planNode instanceof ExchangeNode && ((ExchangeNode) planNode).getScope() == LOCAL)
```

Comment:
nit: you could extract this as a separate RP to reduce number of commits in this PR